domenetjenester/uttak/src/main/java/no/nav/foreldrepenger/domene/uttak/fastsetteperioder/grunnlagbyggere/KontoerGrunnlagBygger.java

```diff
@@ -56,83 +32,38 @@ public KontoerGrunnlagBygger(FagsakRelasjonTjeneste fagsakRelasjonTjeneste,
      * - kan utsettes og  utvide stønadsperioden
      * - Brukes framfor utenAktivitetskravDager fom FAB
      */
-    public Kontoer.Builder byggGrunnlag(UttakInput uttakInput) {
-        var ref = uttakInput.getBehandlingReferanse();
-        var stønadskontoer = hentStønadskontoer(ref);
-        return getBuilder(uttakInput, stønadskontoer)
-            .kontoList(stønadskontoer.stream().filter(k -> k.getStønadskontoType().erStønadsdager()).map(this::map).toList());
+    public Kontoer.Builder byggGrunnlag(UttakInput uttakInput, Stønadskontoberegning stønadskontoberegning) {
+        var kontoer = stønadskontoberegning.getStønadskontoer().stream()
+            .filter(k -> k.getStønadskontoType().erStønadsdager())
+            .map(k -> map(k.getStønadskontoType(), k.getMaxDager()))
+            .toList();
+        return getBuilder(uttakInput, stønadskontoberegning.getStønadskontoutregning()).kontoList(kontoer);
     }
 
-    private Konto.Builder map(Stønadskonto stønadskonto) {
-        return new Konto.Builder().trekkdager(stønadskonto.getMaxDager()).type(UttakEnumMapper.map(stønadskonto.getStønadskontoType()));
+    public Kontoer.Builder byggGrunnlag(UttakInput uttakInput, Map<StønadskontoType, Integer> stønadskontoutregning) {
+        var kontoer = stønadskontoutregning.entrySet().stream()
+            .filter(k -> k.getKey().erStønadsdager())
+            .map(k -> map(k.getKey(), k.getValue()))
+            .toList();
+        return getBuilder(uttakInput, stønadskontoutregning).kontoList(kontoer);
     }
 
-    private Set<Stønadskonto> hentStønadskontoer(BehandlingReferanse ref) {
-        return fagsakRelasjonTjeneste.finnRelasjonFor(ref.saksnummer())
-            .getGjeldendeStønadskontoberegning()
-            .orElseThrow(() -> new IllegalArgumentException("Behandling mangler stønadskontoer"))
-            .getStønadskontoer();
+    private Konto.Builder map(StønadskontoType stønadskonto, Integer maxDager) {
```

Comment:
.kontoList(kontoer)

Litt rart å bruke Konto.Builder her når en ikke bruker builderen til noe? 

Eventuelt bare bruke klassen
public Builder konto(Stønadskontotype konto, int trekkdager) {} i Kontoer.Builder?

For each: builder.konto(k.getType, k.getTrekkdager)?

Reply:
Det er mer finpuss som kan gjøres - men da må man inn i fp-uttak